tools/ports/__init__.py

```diff
@@ -369,14 +369,45 @@ def write_file(filename, contents):
     utils.write_file(filename, contents)
 
 
+class OrderedSet:
+  """Partial implementation of OrderedSet.  Just enough for what we need here."""
+  def __init__(self, items):
+    self.dict = dict()
+    for i in items:
+      self.dict[i] = True
```

Comment:
This could be shortened:
```suggestion
    self.dict = {item: True for item in items}
```

Reply:
Oops, will followup.